nodejs/week2/app.js

```diff
@@ -0,0 +1,99 @@
+const express = require('express')
+const app = express()
+const port = process.env.PORT || 3000
+const documents = require('./documents.json')
+
+// Support parsing JSON requests
+app.use(express.json())
+
+app.get('/', (req, res) => {
+    res.send('This is a search engine')
+})
+
+//GET /search
+app.get('/search', (req,res) => {
+    try {
+        const queryParam = req.query.q;
+        if(queryParam === undefined) {
+            res.send(documents)
+        } else {
+            const resultsArray = searchQuery(queryParam)
+            if (resultsArray.length !== 0) {
+                res.send(resultsArray);
+            } else {
+                res.status(400).send(`There are no items with searched value`);
```

Comment:
404 is more like 'not found'. 400 means bad request if that is what wanted to throw

Reply:
okay